nc-news/src/components/RatingButtonManager.jsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from "react";
+import {
+  ActiveLikeButton,
+  ActiveDislikeButton,
+} from "./styled-components/StyledComponents";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faThumbsUp, faThumbsDown } from "@fortawesome/free-solid-svg-icons";
+
+const likesFromLocalStorage = JSON.parse(
+  localStorage.getItem("likedArticles") || "[]"
+);
+const dislikesFromLocalStorage = JSON.parse(
+  localStorage.getItem("dislikedArticles") || "[]"
+);
+
+export default function RatingButtonManager({ articleId, setVote }) {
+  const [likedArticles, setLikedArticles] = useState(likesFromLocalStorage);
+  const [dislikedArticles, setDislikedArticles] = useState(
+    dislikesFromLocalStorage
+  );
+  const [isLikeButtonActive, setIsLikeButtonActive] = useState(
+    likedArticles.length === 0 ? true : false
+  );
+  const [isDislikeButtonActive, setIsDislikeButtonActive] = useState(
+    dislikedArticles.length === 0 ? true : false
+  );
+
+  // Adding likes to local storage//
+  useEffect(() => {
+    localStorage.setItem("likedArticles", JSON.stringify(likedArticles));
+  }, [isLikeButtonActive]);
+
+  useEffect(() => {
+    localStorage.setItem("dislikedArticles", JSON.stringify(dislikedArticles));
+  }, [isDislikeButtonActive]);
+
+  // HandleClick Event //
+
+  const handleDislikeClick = (inc_vote) => {
+    setVote(inc_vote);
+    if (inc_vote === -1 && !dislikedArticles.includes(articleId)) {
+      setDislikedArticles((currDislikes) => {
+        return [...currDislikes, articleId];
+      });
+      setLikedArticles((currLikes) => {
+        return currLikes.filter((id) => id !== articleId);
+      });
+      setIsDislikeButtonActive(false);
+      setIsLikeButtonActive(true);
+    }
+    if (inc_vote === 1) {
+      setDislikedArticles((currDislikes) => {
+        return currDislikes.filter((id) => id !== articleId);
+      });
+      setIsDislikeButtonActive(true);
+    }
+  };
+
```

Comment:
There's some strange behaviour when you go from upvoting to downvoting and vice versa. For example if you just upvote and it makes the votes 10, if you then immediately downvoted you would expect the total votes to go down by two (undoing the upvote and then downvoting 1). If this is the desired behaviour you can ignore me here!